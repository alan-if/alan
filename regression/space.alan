
Add To Every thing
  Is
    clean.

  Verb clean
    Does Only
      "You clean the"
      If o Is clean Then
        Say o.
      Else 
        Make o clean.
        "dirty $o"
      End If.
      "enthusiastically."
  End Verb.


End Add To.


The aaa Isa location
  
End The aaa.

The gizmo Isa object
  At aaa
  Is 
    Not clean.
  Description
    "There is a"
    If gizmo Is Not clean Then
      "dirty"
    End If.
    "gizmo here."
  
End The gizmo.

Syntax clean = clean (o).

Start At aaa.
  "demonstration of spacing bug"
  "$pa double-quoted string is not started with a space if it is printed
   immediately after a double-quoted string that ended with a dollar-o or
   dollar-1 (and other dollar-things perhaps?)"
  "$p'clean gizmo' to see a missing space before word 'enthusiastically'"