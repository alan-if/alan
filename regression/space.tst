alan space
arun -n space < space.input
