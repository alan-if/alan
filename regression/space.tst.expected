########## space ##########
../compiler/alan space
../interpreter/arun -n space < space.input
########## /bin/sh  space.tst ##########

        No warnings or errors detected.
        2 informational message(s).


demonstration of spacing bug 

a double-quoted string is not started with a space if it is printed
immediately after a double-quoted string that ended with a dollar-o or
dollar-1 (and other dollar-things perhaps?) 

'clean gizmo' to see a missing space before word 'enthusiastically'


Aaa.
There is a dirty gizmo here.

> clean gizmo
You clean the dirty gizmo enthusiastically.

> 

Do you want to RESTART, RESTORE or QUIT ? 
